File: C++/598.cpp
__________________________________________________________________________________________________
sample 8 ms submission
static int x = []() {
    std::ios::sync_with_stdio(false);
    cin.tie(NULL);
    return 0; 
}();

class Solution {
public:
    int maxCount(int m, int n, vector<vector<int>>& ops) {
        int min_m = m;
        int min_n = n;
        for(int i=0;i<ops.size();i++){
            if(min_m>ops[i][0]) min_m = ops[i][0];
            if(min_n>ops[i][1]) min_n = ops[i][1];
        }
        
        return min_m*min_n;
    }
};
__________________________________________________________________________________________________
sample 11796 kb submission
class Solution {
public:
    int maxCount(int m, int n, vector<vector<int>>& ops) {
        int x = m;
        int y = n;
        for (int i = 0; i < ops.size(); i++) {
            x = x < ops[i][0] ? x : ops[i][0];
            y = y < ops[i][1] ? y : ops[i][1];
        }
        return x*y;
    }
};
__________________________________________________________________________________________________

File: Java/598.java
__________________________________________________________________________________________________
sample 0 ms submission
class Solution {
    // find minimum i and j from ops, then multiply
    public int maxCount(int m, int n, int[][] ops) {
        int iMin = m;
        int jMin = n;
        for (int[] op : ops) {
            if (op[0] < iMin) iMin = op[0];
            if (op[1] < jMin) jMin = op[1];
        }
        return iMin * jMin;
    }
}
__________________________________________________________________________________________________
sample 35104 kb submission
class Solution {
    public int maxCount(int m, int n, int[][] ops) {
        if (ops == null || ops.length == 0) return m*n;
        Arrays.sort(ops, (o1, o2) -> Integer.compare(o1[0], o2[0]));
        int r = ops[0][0];
        Arrays.sort(ops, (o1, o2) -> Integer.compare(o1[1], o2[1]));
        int c = ops[0][1];
        return r * c;
    }
}
__________________________________________________________________________________________________
